gossipd/routing.c

```diff
@@ -636,19 +726,20 @@ const struct short_channel_id *handle_channel_announcement(
 		     type_to_string(pending, struct short_channel_id,
 				    &pending->short_channel_id));
 
-	/* FIXME: Handle duplicates as per BOLT #7 */
+	/* So you're new in town, ey? Let's find you a room in the Inn. */
+	if (chan == NULL)
+		chan = routing_channel_new(chan, &pending->short_channel_id);
+	chan->pending = pending;
```

Comment:
Would it be better for `chan` to `tal_steal` the `pending` at this point?  After all, `rstate` has not direct pointer to the `pending`.